cmd/snap/cmd_run.go

```diff
@@ -118,6 +118,37 @@ func getSnapInfo(snapName string, revision snap.Revision) (*snap.Info, error) {
 	return info, nil
 }
 
+func createOrUpdateUserDataSymlink(info *snap.Info, usr *user.User) error {
+	// 'current' symlink for user data (SNAP_USER_DATA)
+	userData := info.UserDataDir(usr.HomeDir)
+	wantedSymlinkValue := filepath.Base(userData)
+	currentActiveSymlink := filepath.Join(userData, "..", "current")
+
+	var err error
+	var currentSymlinkValue string
+	for i := 0; i < 5; i++ {
+		currentSymlinkValue, err = os.Readlink(currentActiveSymlink)
+		// failure other than non-existing symlink is fatal
+		if err != nil && !os.IsNotExist(err) {
+			// TRANSLATORS: %v the error message
+			return fmt.Errorf(i18n.G("cannot read symlink: %v"), err)
```

Comment:
If the symlink does not exist then let's just do `reschedule and continue` like I recommended below.